run_validate_drp.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+set -eo pipefail
+
+print_error() {
+  >&2 echo -e "$@"
+}
+
+fail() {
+  local code=${2:-1}
+  [[ -n $1 ]] && print_error "$1"
+  # shellcheck disable=SC2086
+  exit $code
+}
+
+
+check_env_vars() {
+  local req_vars=(
+    LSST_VALIDATE_DRP_DATASET
+    LSST_VALIDATE_DRP_DATASET_DIR
+  )
+
+  local err
+  for v in "${req_vars[@]}"; do
+    [[ ! -v $v ]] && err="${err}Missing required env variable: ${v}\\n"
+  done
+
+  [[ -n $err ]] && fail "$err"
+
+  return 0
+}
+
+find_mem() {
+  # Find system available memory in GiB
+  local os
+  os=$(uname)
+
+  local sys_mem=""
+  case $os in
+    Linux)
+      [[ $(grep MemAvailable /proc/meminfo) =~ \
+         MemAvailable:[[:space:]]*([[:digit:]]+)[[:space:]]*kB ]]
+      sys_mem=$((BASH_REMATCH[1] / 1024**2))
+      ;;
+    Darwin)
+      # I don't trust this fancy greppin' an' matchin' in the shell.
+      local free
+      free=$(vm_stat | grep 'Pages free:' | tr -c -d '[:digit:]')
+      local inac
+      inac=$(vm_stat | grep 'Pages inactive:' | tr -c -d '[:digit:]')
+      sys_mem=$(( (free + inac) / ( 1024 * 256 ) ))
+      ;;
+    *)
+      >&2 echo "Unknown uname: $os"
+      exit 1
+      ;;
+  esac
+
+  echo "$sys_mem"
+}
+
+# find the maximum number of processes that may be run on the system
+# given the the memory per core ratio in GiB -- may be expressed in
+# floating point.
+target_cores() {
+  local mem_per_core=${1:-1}
+
+  local sys_mem
+  sys_mem=$(find_mem)
+  local sys_cores
+  sys_cores=$(getconf _NPROCESSORS_ONLN)
+
+  # bash doesn't support floating point arithmetic
+  local target_cores
+  #target_cores=$(echo "$sys_mem / $mem_per_core" | bc)
```

Comment:
I don't know why this doesn't work, but should probably be removed if you are using `awk` instead.

Reply:
Fair enough -- removed.